pkg/crd/crd.go

```diff
@@ -58,17 +58,26 @@ var (
 // Apply will create or update all project CRDs inside a Kubernetes cluster.
 //
 // The latest available version of the CRD will be used to perform this operation.
-func Apply(ctx context.Context) error {
-	return operate(ctx, applyFn)
+func Apply(ctx context.Context, istioEnabled bool) error {
+	return operate(ctx, applyFn, istioEnabled)
 }
 
 // Delete will remove all project CRDs from a Kubernetes cluster.
-func Delete(ctx context.Context) error {
-	return operate(ctx, deleteFn)
+func Delete(ctx context.Context, istioEnabled bool) error {
+	return operate(ctx, deleteFn, istioEnabled)
 }
 
 // operate will read all available CRDS and apply state changes to the cluster using the processor func.
-func operate(ctx context.Context, processor crdProcessor) error {
+func operate(ctx context.Context, processor crdProcessor, istio bool) error {
+	if istio {
+		quit, err := waitForIstioSidecar()
+		if err != nil {
+			return err
+		}
+
+		defer quit()
```

Comment:
You need `quit` to run regardless of `err`, no? Otherwise it'll just hang?

Reply:
Hmm, is that true? The only reason I can see an `err` occurring here is if we fail to establish the Istio sidecar's health in the first place. If we can't assert Istio's health, can we then tell Istio to quit?

Reply:
Fair point